mod-clans/src/main/java/gg/solarmc/loader/clans/ClanManager.java

```diff
@@ -0,0 +1,178 @@
+/*
+ *
+ *  * dataloader
+ *  * Copyright © 2021 SolarMC Developers
+ *  *
+ *  * dataloader is free software: you can redistribute it and/or modify
+ *  * it under the terms of the GNU Affero General Public License as
+ *  * published by the Free Software Foundation, either version 3 of the
+ *  * License, or (at your option) any later version.
+ *  *
+ *  * dataloader is distributed in the hope that it will be useful,
+ *  * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  * GNU Affero General Public License for more details.
+ *  *
+ *  * You should have received a copy of the GNU Affero General Public License
+ *  * along with dataloader. If not, see <https://www.gnu.org/licenses/>
+ *  * and navigate to version 3 of the GNU Affero General Public License.
+ *
+ */
+
+package gg.solarmc.loader.clans;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import gg.solarmc.loader.SolarPlayer;
+import gg.solarmc.loader.Transaction;
+import gg.solarmc.loader.data.DataManager;
+import gg.solarmc.loader.schema.tables.records.ClansClanAlliancesRecord;
+import gg.solarmc.loader.schema.tables.records.ClansClanInfoRecord;
+import org.jooq.DSLContext;
+
+import java.time.Duration;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static gg.solarmc.loader.schema.tables.ClansClanAlliances.CLANS_CLAN_ALLIANCES;
+import static gg.solarmc.loader.schema.tables.ClansClanInfo.CLANS_CLAN_INFO;
+import static gg.solarmc.loader.schema.tables.ClansClanMembership.CLANS_CLAN_MEMBERSHIP;
+
+
+public class ClanManager implements DataManager {
+
+    private final Cache<Integer,Clan> clans = Caffeine.newBuilder().expireAfterAccess(Duration.ofMinutes(10)).build();
+    private final Cache<Integer,Integer> allianceCache = Caffeine.newBuilder().build();
+
+    /**
+     * Gets a clan from memory cache or from table if not present.
+     * @param transaction the tx
+     * @param id ID of clan requested
+     * @return the clan you asked for idiot
+     * @throws IllegalStateException if the clan isn't present in table
+     */
+    public Clan getClan(Transaction transaction, Integer id) {
```

Comment:
This method should probably be package-private. If not, it should definitely declare `int id`